tests/stack/test_stack_run.py

```diff
@@ -0,0 +1,104 @@
+import unittest
+
+import netsquid as ns
+from netqasm.sdk import Qubit
+from netsquid_netbuilder.modules.qdevices.generic import GenericQDeviceConfig
+from netsquid_netbuilder.modules.qlinks.depolarise import DepolariseQLinkConfig
+from netsquid_netbuilder.util.network_generation import create_2_node_network
+
+from squidasm.run.stack.run import run
+from squidasm.sim.stack.program import Program, ProgramContext, ProgramMeta
+from squidasm.util.routines import teleport_recv, teleport_send
+
+
+class TeleportSenderProgram(Program):
+    def __init__(self, peer_name: str):
+        self.peer_name = peer_name
+        self.complete = False
+
+    @property
+    def meta(self) -> ProgramMeta:
+        return ProgramMeta(
+            name="sender_program",
+            csockets=[self.peer_name],
+            epr_sockets=[self.peer_name],
+            max_qubits=2,
+        )
+
+    def run(self, context: ProgramContext):
+        q = Qubit(context.connection)
+        yield from teleport_send(q, context, peer_name=self.peer_name)
+        self.complete = True
+
+
+class TeleportReceiverProgram(Program):
+    def __init__(self, peer_name: str):
+        self.peer_name = peer_name
+        self.complete = False
+
+    @property
+    def meta(self) -> ProgramMeta:
+        return ProgramMeta(
+            name="receiver_program",
+            csockets=[self.peer_name],
+            epr_sockets=[self.peer_name],
+            max_qubits=2,
+        )
+
+    def run(self, context: ProgramContext):
+        yield from teleport_recv(context, peer_name=self.peer_name)
+        self.complete = True
+
+
+class TestRun(unittest.TestCase):
```

Comment:
I don't see how this test case is relevant for the S and T gates.... maybe it  tests the bugfix also in this PR?

Reply:
This is used to check the `run` method. This is a testcase covering the simulation time issue